src/init/mod.rs

```diff
@@ -180,23 +180,21 @@ fi"#,
             );
             Some(script)
         }
-        None => {
-            println!(
-                "Invalid shell name provided: {}\\n\
-                 If this issue persists, please open an \
-                 issue in the starship repo: \\n\
-                 https://github.com/starship/starship/issues/new\\n\"",
-                shell_name
-            );
-            None
-        }
-        Some(shell_basename) => {
+        _ => {
             println!(
                 "printf \"\\n{0} is not yet supported by starship.\\n\
-                 For the time being, we support bash, zsh, fish, and ion.\\n\
+                 For the time being, we support the following shells:\\n\
+                 * bash\\n\
+                 * elvish\\n\
+                 * fish\\n\
+                 * ion\\n\
+                 * powershell\\n\
+                 * tcsh\\n\
+                 * zsh\\n\
+                 \\n\
                  Please open an issue in the starship repo if you would like to \
-                 see support for {0}:\\nhttps://github.com/starship/starship/issues/new\"\\n\\n",
-                shell_basename
+                 see support for {0}:\\nhttps://github.com/starship/starship/issues/new\\n\\n\"",
+                shell_name
             );
```

Comment:
I think this will use for escaping (this will also get rid of any format string vulnerabilities).
```rust
            let quoted_arg = shell_words::quote(shell_name);
            println!(
                "printf \"\\n%s is not yet supported by starship.\\n\
                 For the time being, we support the following shells:\\n\
                 * bash\\n\
                 * elvish\\n\
                 * fish\\n\
                 * ion\\n\
                 * powershell\\n\
                 * tcsh\\n\
                 * zsh\\n\
                 \\n\
                 Please open an issue in the starship repo if you would like to \
                 see support for %s:\\nhttps://github.com/starship/starship/issues/new\\n\\n\" {0} {0}",
                 quoted_arg
            );
```

Re: combining `None` and `Some(unknown)`, you could also just ` let shell_basename = Path::new(shell_name).file_stem().and_then(OsStr::to_str).unwrap_or(shell_name);` to get rid of the `Option<_>`.